hub/scripts.js

```diff
@@ -9,258 +9,338 @@
-          content: value,
-        });
-        // description to fall through to default
-        // eslint-disable-next-line no-fallthrough
-      default:
-        md.push({
-          name: key,
-          content: value,
-        });
+    // Make the string lower case;
+    // then transform all the special characters to '-';
+    // a longer sequence of special characters will be replaced by a single '-'
+    return str.toLowerCase().replace(/[^0-9a-z]{1,}/gi, '-');
+  }
+
+  /**
+   * Creates and returns an element based on a provided tag name and attributes
+   * @param {String} tagName The tag name of the desired element
+   * @param {Object} attributes Object containing key/value pairs
+   * for the attributes to be attached to the created element
+   * @return {Element} The resulting HTML element
+   */
+  function createCustomElement(tagName, attributes) {
+    // If no tag name is provided, no element is created
+    if (!isNonEmptyString(tagName)) {
+      return undefined;
     }
-  });
-  const $tags = Array.from(document.head.querySelectorAll('meta'));
-  const $frag = document.createDocumentFragment();
-  md.forEach((m) => {
-    const $tag = $tags.find((t) => t.getAttribute('name') === m.name || t.getAttribute('property') === m.property);
-    if ($tag) {
-      // update existing meta tag
-      $tag.setAttribute('content', m.content);
-    } else {
-      // add new meta tag
-      $frag.appendChild(createTag('meta', m));
+
+    const element = document.createElement(tagName);
+
+    if (typeof attributes === 'object') {
+      for (const [key, value] of Object.entries(attributes)) {
+        if (isNonEmptyString(key)) {
+          // The value needs to be a string. If another type
+          // is used, it will be transformed into an empty string
+          const parsedValue = !isNonEmptyString(value) ? '' : value;
+
+          element.setAttribute(key, parsedValue);
+        }
+      }
     }
-  });
-  if ($frag.childNodes.length) {
-    document.head.appendChild($frag);
+
+    return element;
   }
-  $metaBlock.remove();
-}
 
-/**
- * Turn tables to DIV.
- * @param {object} $table Table element
- */
-function tableToDivs($table) {
-  const $rows = $table.querySelectorAll('tbody tr');
-  const blockname = $table.querySelector('th').textContent;
-  const $block = createTag('div', { class: `${toClassName(blockname)}` });
-  $rows.forEach(($tr) => {
-    const $row = createTag('div');
-    $tr.querySelectorAll('td').forEach(($td) => {
-      const $div = createTag('div');
-      $div.innerHTML = $td.innerHTML;
-      $row.append($div);
+  /**
+   * Reads key/value pairs from a block.
+   * @param {HTMLElement} $block The block
+   * @returns {Object} The configuration
+   */
+  function readBlockConfig($block) {
+    const config = {};
+    $block.querySelectorAll(':scope>div').forEach(($row) => {
+      if ($row.children && $row.children[1]) {
+        const name = stringToClassname($row.children[0].textContent);
```

Comment:
again, future improvement: `$row.children[0]` and `$row.children[1]` can be cached